atom/browser/atom_download_manager_delegate.cc

```diff
@@ -154,6 +179,18 @@ void AtomDownloadManagerDelegate::OnDownloadPathGenerated(
       file_type_info.extensions.push_back(extensions);
     }
     file_type_info.include_all_files = true;
+
+    // To show file save dialog, |window| always must be valid.
+    NativeWindow* window = nullptr;
+    if (content::WebContents* web_contents = item->GetWebContents()) {
+      window = GetNativeWindowFromWebContents(web_contents);
+      CloseWebContentsIfNeeded(web_contents);
+    }
+
+    if (!window)
+      window = GetLastActiveWindow();
```

Comment:
This can be removed. The reason of the crash is inappropriate closing webcontent at browser-laptop

Reply:
The intention for that change is making it safer than now for null window situation.
If the download is started from a webcontent that isn't used for tab but for download, that null case might happen.

In any case, if you think that is a situation that is never occurring or might have any other undersired effect, we can remove it.

Reply:
wdyt @bridiver ?

Reply:
I don't think this is necessary because it doesn't cover all the edge cases anyway (no windows at all). If the webcontents window is null I think we should just abort the download

Reply:
Instead of abort download, how about downloading with default target path?
We can proceed file download with DownloadManager if we can't determine target path in some situation such as file save dialog cannot visible.